src/Util/AstHelper.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Util;
+
+use Closure;
+use PhpParser\Node;
+
+use Roave\BetterReflection\BetterReflection;
+use Roave\BetterReflection\Reflection\Reflection;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionFunction;
+
+use PhpParser\Node\Expr\Closure as ClosureNode;
+use PhpParser\Node\Expr\ClosureUse as ClosureUseNode;
+use PhpParser\Node\Expr\ArrowFunction as ArrowFunctionNode;
+
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor\FindingVisitor;
+use PhpParser\PrettyPrinter\Standard as StandardPrettyPrinter;
+use PhpParser\PrettyPrinterAbstract;
+use RuntimeException;
+
+final class AstHelper {
+    private BetterReflection $betterReflection;
+
+    public function __construct( ?BetterReflection $betterReflection = null ) {
+        if ( !$betterReflection  ) {
+            $this->betterReflection = new BetterReflection();
+        } else {
```

Comment:
I suggest we always ask for dependencies, especially in new code.

We usually make these optional only to help where the API wasn't well defined upfront, and we needed to preserve BC.